### ISSUE TITLE
desktop macOS security permission

### ISSUE THREAD

**Issue by exalted** [zed-industries/zed#5138]
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

If I previously didn't allow Zed to access my Desktop (i.e. macOS System Settings -> Privacy & Security -> Files and Folders -> Zed -> Desktop Folder [disabled]), when opening a file on the Desktop via the command line (e.g. `zed ~/Desktop/example.txt`) I'll see the file on tree view on Zed UI, however, on the command line there's an error message saying `error opening "/Users/<username>/Desktop/example.txt": Operation not permitted (os error 1)` and the file contents won't load on Zed UI.

### Describe the solution you'd like

I've only tried via the command line, so I am not sure if the UX is any better via `File > Open` via the UI; either way, I guess Zed UI (and the command line, for that matter) could give a more user-friendly error message, something like "Click here to open System Settings and allow Zed to access to your Desktop"…

**Comment by Blankeos** [comment]
I have a same error of trying to monkeypatch an /Applications/*.js file. I can't edit it with Zed. There's no option to allow it either checking from Mac's settings.

**Comment by notpeter** [comment]
Yep. It would seem reasonable to try and detect when we're being blocked by macOS permissions (`Operation not permitted`) and link them to the right place:
```sh
open "x-apple.systempreferences:com.apple.preference.security?Privacy_FilesAndFolders"
```

**Comment by imumesh18** [comment]
There is another bug which I caught when I tried to make a fix to zed ask for permission in this case through cli.

Bug: If you try to open a file from ui with open select a file which doesn't have permission deny permission. In this case the zed shouldn't open the file. But it seems the file opens and I am allowed to make changes to file as well I tested this with stable and same behaviour exists.

![Image](https://github.com/user-attachments/assets/b2ab99ac-5713-4b1b-bc97-c638689ddc06)

Edit: seems to be the same behaviour in vscode as well. Not sure if this a bug or expected behaviour at this point.
